include/RAJA/pattern/kernel/internal.hpp

```diff
@@ -278,6 +281,21 @@ RAJA_INLINE RAJA_HOST_DEVICE void invoke_lambda(Data &&data)
       std::forward<Data>(data));
 }
 
+//Lambda with custom arguments
+RAJA_SUPPRESS_HD_WARN
+template<camp::idx_t LoopIndex, typename Data, typename T, camp::idx_t... Args>
+RAJA_INLINE RAJA_HOST_DEVICE void invoke_custom_lambda(Data &&data,T myTuple)
+{
+  camp::get<LoopIndex>(data.bodies)(camp::get<Args>(myTuple)...);
```

Comment:
I think if you merge in the call_extractor logic here, you'll never have to build the tuple in the first place... the argument would just drop into the expansion

Reply:
Here I got a bit stuck. I wasn't sure how to further merge the code together.

Reply:
Here too, there's nothing wrong with this or call_extractor, but we might be able to make it a bit easier to follow.